packages/test/stochastic-test-utils/src/performActions.ts

```diff
@@ -219,20 +226,27 @@ export function performFuzzActions<
 
 	for (let operation = generator(state); operation !== done; operation = generator(state)) {
 		operations.push(operation);
-		if (saveInfo !== undefined && operations.length === saveInfo.saveAt) {
-			writeFileSync(saveInfo.filepath, JSON.stringify(operations));
+		if (saveInfo?.saveWhenFinish === false && operations.length === saveInfo.saveAt) {
+			writeFileSync(saveInfo.filepath, JSON.stringify(operations, undefined, 4));
```

Comment:
@Abe27342 - what do you think of the added indentation here? Is the added file size likely to be an issue? If so, we could make it configurable.